cranelift-codegen/src/regalloc/reload.rs

```diff
@@ -211,6 +211,46 @@ impl<'a> Context<'a> {
         debug_assert!(self.candidates.is_empty());
         self.find_candidates(inst, constraints);
 
+        // If we find a copy from a stack slot to the same stack slot, replace
+        // it with a `copy_nop` but otherwise ignore it.  In particular, don't
+        // generate a reload immediately followed by a spill.  The `copy_nop`
+        // has a zero-length encoding, so will disappear at emission time.
+        if let InstructionData::Unary {
+            opcode: Opcode::Copy,
+            arg,
+        } = self.cur.func.dfg[inst]
+        {
+            let dst_vals = self.cur.func.dfg.inst_results(inst);
+            if dst_vals.len() == 1 {
+                let dst_val = dst_vals[0];
+                let src_loc = self.cur.func.locations[arg];
+                let dst_loc = self.cur.func.locations[dst_val];
+                let src_aff = self.liveness[arg].affinity;
+                let dst_aff = self.liveness[dst_val].affinity;
+                let can_transform = match (src_loc, dst_loc) {
```

Comment:
nit: since `src_loc`, `dst_loc`, `src_aff`, `dst_aff` are used only once, can you replace their uses with their definitions, please? (or does the borrow check not like that)